### DISCUSSION TITLE
Control of an IR emitter transistor

### DISCUSSION THREAD

**Discussion by robzed** [tabemann/zeptoforth#71] [Show and tell]
Control of an IR emitter transistor.
![image](https://github.com/tabemann/zeptoforth/assets/4871013/3c929878-1f08-47af-8181-4ba450d09e74)

Ran by this code: 
https://github.com/robzed/ukmarsbot_forth/blob/master/for_rp2040/robot-adc.fth

line 145 turns on the IR LED
        dup EMITTER_B pin!
This is the rising edge on the image above.

Then we ask the hardware to read 4 ADC channels and interrupt us when the FIFO is full.

On the next interrupt, line 151 turns off the IR LED
        false EMITTER_B pin!
This is the falling edge on the image above. 

From rising to falling is about 25 us.

We repeat this sequence every 2ms. 

Interrupt latency isn't a problem here.

**Comment by tabemann** [comment]
Nice!
